File: go/pascals-triangle/pascals_triangle.go
// Paascal's Triangle
package pascalstriangle

// Return Pascal's Triangle up to row n.
// @param n: The row to stop on
// @returns: Slice of Slices of Integer where each sub-slice is a row
// of Pascal's Triangle.
func Triangle(n int) [][]int {
	var ret [][]int = make([][]int, n)
	for i := 1; i <= n; i++ {
		count := 1
		switch i {
		case 1:
			ret[i-1] = []int{1}
		case 2:
			ret[i-1] = []int{1, 1}
		default:
			previous := ret[i-2]
			count = len(previous) + 1
			row := make([]int, count)
			row[0] = previous[0]
			row[count-1] = previous[len(previous)-1]
			for i := 0; i < len(previous)-1; i++ {
				row[i+1] = previous[i] + previous[i+1]
			}
			ret[i-1] = row
		}
	}
	return ret
}

File: php/pascals-triangle/PascalsTriangle.php
<?php
// Pascal's Triangle exercise

declare(strict_types=1);

// generate Pascal's Triangle. Each row of the triangle is an array with the numbers in that row.
// will only generate up to the give $rowCount.
// @param $rowCount: How many rows of Pascal's triangle to generate.
// @returns: Array with each element another array with the values for that line of Pascal's triangle.
function pascalsTriangleRows(int $rowCount)
{
    $result = array();
    if ($rowCount < 1) {
        return $result;
    }
    for($i = 0; $i < $rowCount; $i++) {
        if ($i == 0) {
            $result[] = array(1);
        } else {
            $previous = $result[$i - 1];
            $current = array();
            $current[] = $previous[0];
            for($j = 0; $j < count($previous) - 1; $j++) {
                $current[] = $previous[$j] + $previous[$j + 1];
            }
            $current[] = $previous[count($previous) - 1];
            $result[] = $current;
        }
    }
    return $result;
}
